src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,41 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Your login can not be null. Please, enter a valid login");
+        } else if (user.getLogin().isEmpty()) {
+            throw new RuntimeException("Your login can not be empty. Please, enter a valid login");
+        } else if (user.getLogin().isBlank()) {
+            throw new RuntimeException("Your login can not contain blank spaces only."
+                    + " Please, enter a valid login");
+        } else if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("This user is already exist");
+        } else if (user.getAge() == null) {
+            throw new RuntimeException("The age can not be null. Please, enter your age");
```

Comment:
the same, maybe let's create one if statement of all null values, and add here null check from below for other fields. But its up to you
```suggestion
        } else if (user.getAge() == null || user.getPassword() == null) {
            throw new RuntimeException("You have empty lines. Please, fill in all lines");
```